src/service/github-allocator-registry/github-allocator-registry.service.ts

```diff
@@ -0,0 +1,146 @@
+    const appOctokit = new Octokit({
+      authStrategy: createAppAuth,
+      auth: {
+        appId: this.configService.get<string>('GITHUB_APP_ID'),
+        privateKey: this.configService.get<string>('GITHUB_PRIVATE_KEY'),
+      },
+    });
+
+    const response = await appOctokit.request(
+      'GET /repos/{owner}/{repo}/installation',
+      {
+        owner: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_OWNER'),
+        repo: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_NAME'),
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      },
+    );
+
+    return response.data.id;
+  }
+
+  private async getOctokit(): Promise<Octokit> {
+    if (!this.octokit) {
+      this.octokit = new Octokit({
+        authStrategy: createAppAuth,
+        auth: {
+          appId: this.configService.get<string>('GITHUB_APP_ID'),
+          privateKey: this.configService.get<string>('GITHUB_PRIVATE_KEY'),
+          installationId: await this.getInstallationId(),
+        },
+      });
+    }
+    return this.octokit;
+  }
+
+  public async getAllocatorsRegistry(): Promise<AllocatorRegistry[]> {
+    const octokit = await this.getOctokit();
+    const response = (await octokit.request(
+      'GET /repos/{owner}/{repo}/contents/{path}',
+      {
+        owner: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_OWNER'),
+        repo: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_NAME'),
+        path: 'Allocators',
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      },
+    )) as any;
+
+    const paths = response.data
+      .filter(
+        (v) => v.name.endsWith('.json') && v.name != 'Allocator JSON SPEC.json',
+      )
+      .map((v) => v.path);
+
+    const registry = [];
+    for (const path of paths) {
+      try {
+        const info = await this.getAllocatorInfo(path);
+        if (info) registry.push(info);
+      } catch (err) {
+        this.logger.error(
+          `Error while fetching registry info for ${path}: ${err.message}`,
+          err.cause || err.stack,
+        );
+      }
+    }
+    return registry;
+  }
+
+  private async getAllocatorInfo(
+    path: string,
+  ): Promise<AllocatorRegistry | null> {
+    const octokit = await this.getOctokit();
+    const file = (await octokit.request(
+      'GET /repos/{owner}/{repo}/contents/{path}',
+      {
+        owner: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_OWNER'),
+        repo: this.configService.get<string>('ALLOCATOR_REGISTRY_REPO_NAME'),
+        path: path,
+        headers: {
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+      },
+    )) as any;
+
+    const data = JSON.parse(atob(file.data.content));
+    if (!data?.pathway_addresses?.msig) {
+      this.logger.warn(`No msig address for ${path}`);
+      return;
+    }
+
+    const id = await this.lotusApiService.getFilecoinClientId(
```

Comment:
I have a problem with this method - it says it's fetching **client** id and it stores it in a `client_address_mapping` cache table, but it isn't really specific to clients in any way.
Do we leave it like this or de we refactor it and migrate the db to use something like `id_address_mapping` instead?

Reply:
So `Filecoin.StateLookupID` works for allocators as well? This is very confusing, I vote for `id_address_mapping` refactoring.

Reply:
> So Filecoin.StateLookupID works for allocators as well? 

It works for all Filecoin accounts, doesn't matter if they're a miner, allocator, client or nobody at all.

I will do the refactor.

Reply:
https://github.com/fidlabs/compliance-data-platform/pull/87/commits/3406cf04cfbdde7cad82cbca2c250fff299b3c24
Done.